operators/src/engine/execution_context.rs

```diff
@@ -59,6 +59,8 @@ pub trait ExecutionContext: Send
     async fn read_ml_model(&self, path: PathBuf) -> Result<String>;
 
     async fn write_ml_model(&mut self, path: PathBuf, ml_model_str: String) -> Result<()>;
+
+    async fn resolve_named_data(&self, data: &NamedData) -> Result<DataId>;
```

Comment:
as discussed, I think we can remove this method and have the `MetaDataProvider`s `meta_data` method just accept a `NamedData` directly, if it is able to resolve it anyway.

On the other hand it would also be neat, if the `NamedData` was only a thing in the services crate and the other crates just worked wit `DataId`s but I guess this is not really feasible?

Reply:
We cannot do this, since it is also part of operators, which are also part of our API.
We should work on our API encapsulation. Then, we could do this.